### DISCUSSION TITLE
Page not hydrating in prod mode but works fine in dev mode

### DISCUSSION THREAD

**Discussion by mlynch** [vercel/next.js#18428] [Help] [answered]
Hello,

I've been struggling to nail down an issue where my Next.js app (w/ typescript) works fine in dev mode but doesn't hydrate (no event handlers, etc.) in prod mode.

Unfortunately I get no build errors or any runtime errors. I was able to reliably fix/repro the issue after reducing my app down to a single internal import, but the import is just a simple function, so I'm completely stumped.

I'm using a fairly standard Next.js setup, no custom webpack config like the poster [in this issue](https://github.com/vercel/next.js/discussions/16053).

Here is the [repo](https://github.com/mlynch/init-app) with a contrived [index.tsx](https://github.com/mlynch/init-app/blob/main/pages/index.tsx) where the button works fine in dev mode but not in prod mode. 

Any help greatly appreciated!

**Comment by jamesmosier** [comment, marked as answer]
Hi @mlynch. I cloned your repo and wasn't able to reproduce 🤔  I know that's not super helpful, but maybe because I was coming in with a fresh build and `node_modules` that helped out in my case 🤷‍♂️

**Reply by mlynch** [reply]
Wow, well that's encouraging. So you cloned the repo, ran `npm install`, then `npm run build && npm run start` or something to that effect? 

And when you clicked the button you saw this output:

<img width="165" alt="image" src="https://user-images.githubusercontent.com/11214/97508407-89341480-194d-11eb-9db1-aa83ec5048fc.png">

I just tried this again, fresh clone and install and the above, and when I click the button nothing happens (both FF and Chrome). Works fine in `npm run dev` though.

Hmmmm 🤔 I wonder if something is using a global dependency and that's the difference? Perhaps TypeScript?

**Reply by jamesmosier** [reply]
Oh actually, I did do one thing differently: I used yarn! Therefore it did not use your `package-lock` file. So I just reran it using `npm` instead and saw the page **not** working! So it appears a dependency in your `package-lock.json` is potentially causing the issue.

**Reply by mlynch** [reply]
Interesting! I tried it w/ yarn and it works fine. I killed my package lock and `node_modules` and tried `npm` again and it's broken. Very strange.

Thanks for the help by the way, really appreciate it!

**Reply by mlynch** [reply]
Haven't found the culprit but going to stick with yarn for now. Makes sense this would be a dep issue. I've got the output of `npm ls` and `yarn list` but will dig through that if need be.

Thanks again, this got me pointed in the right direction 👍
